### DISCUSSION TITLE
2 JPG Images, overlaying one on another, the background one with opacity

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#5210] [Development]
magick -version
Version: ImageMagick 7.1.0-35 Q16-HDRI x86_64 20062 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): bzlib djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lzma pangocairo png tiff webp x xml zlib
Compiler: gcc (8.5)

I have 2 images. One of them I want to set as background with opacity. The other one, I want to make it transparent.

magick \( 793.jpg -resize 3500x -alpha Set -background white -channel A -evaluate Multiply 0.05 +channel \) -trim +repage \( 00012-watermark.jpg -fuzz 5%% -background none -transparent white -morphology erode diamond:1 \) -gravity east -geometry +200+0 -compose Over -composite out.jpg

I am not getting the opacity for the background image.

Thanks,

![793](https://user-images.githubusercontent.com/12046302/172766599-79e8119f-12fb-4f61-9267-c4e2b9e762f1.jpg)
![00012-watermark](https://user-images.githubusercontent.com/12046302/172767061-c85486ee-c4b0-40c4-9963-f6447b230351.jpg)
![out](https://user-images.githubusercontent.com/12046302/172766805-982f5d7c-6a7d-4c98-b564-40f64ecee102.jpg)

**Comment by fmw42** [comment]
Your result looks fine. What is wrong with it? The white was turned transparent or it would show.

Does this do what you want? Unix syntax for IM 7

```
magick background.jpg \
-set option:dims "%wx%h" \
\( foreground.jpg -resize "%[dims]" -fuzz 3% -transparent white \) \
-gravity southeast -compose over -composite \
result.jpg

```

![result](https://user-images.githubusercontent.com/13458218/172770056-1af95d9a-0373-45d8-b2d8-262989c01b95.jpg)

**Comment by snibgo** [comment]
@srini8080: you write the output to JPG, which cannot record transparency. Therefore the "-channel A -evaluate Multiply 0.05 +channel" on the first input has no effect.

**Reply by srini8080** [reply]
Good Point. So what is the best way? First make the background into an image with reduced opacity and then put the front image on the top of it? I need to make a jpg for the final combined image. 

Thanks,

**Comment by snibgo** [comment]
I don't understand what you mean by "reduced opacity". Reducing the opacity does nothing if the opacity isn't recorded.

Perhaps you mean that you want the background to become lighter, or something. Windows BAT syntax:
```
magick ^
  ( 793.jpg -resize 3500x -alpha Set -background white -channel A -evaluate Multiply 0.40 +channel ) ^
  -trim +repage ^
  -layers Flatten ^
  ( 00012-w.jpg -fuzz 5%% -background none -transparent white -morphology erode diamond:1 ) ^
  -gravity east -geometry +200+0 ^
  -compose Over -composite ^
  out3.jpg
```
I have adjusted the alpha multiplier so we can still see the image, and flattened the background against white.
![out3](https://user-images.githubusercontent.com/33812368/172821603-d8fe61b8-9b92-4bda-a07a-19f27a21cd9d.jpg)

**Reply by srini8080** [reply]
Sorry, wrong technical term.

Fantastic. This is what I was looking for.

Thanks much

**Comment by Zeioth** [comment]
What would be the bash syntax for this? I don't seem to be able to get it right. I'm trying to have input.jpg > overlay.jpg > texture.png where texture.png is mostly transparent.

```
magick \
  input.png \
  -trim +repage \
  -layers Flatten \
  overlay.png texture.png -alpha Set -background white -channel A -evaluate Multiply 0.40 +channel \
  -gravity northwest -geometry +436+277 \
  -compose Over -composite \
  ./output.jpg
```

This is my code, but it doesn't work correctly.

**Reply by fmw42** [reply]
I am not sure what you are asking. But as a guess, try

```
magick \
  input.png \
  -trim +repage \
   \( overlay.png texture.png -alpha Set -background white -channel A -evaluate Multiply 0.40 +channel \) \
  -gravity northwest -geometry +436+277 \
  -compose Over -composite \
  ./output.jpg
```

**Reply by Zeioth** [reply]
Thank you for your answer! I managed to do exactly what I wanted in a simple way, but in two steps:

    magick convert input.png overlay.png -gravity northwest -geometry +636+277 -composite tmp.png
    magick convert tmp.png texture.png -gravity northwest -geometry +636+277 -composite output.jpg

There some way to do this in only one command?

**Comment by fmw42** [comment]
First, in IM 7, use magick, not magick convert. Just string the two commands together

```
magick input.png overlay.png -gravity northwest -geometry +636+277 -composite \
texture.png -gravity northwest -geometry +636+277 -composite output.jpg
```

**Reply by Zeioth** [reply]
Oh my god it's true. Thank you so much!!!
